crt_portal/cts_forms/views.py

```diff
@@ -155,6 +153,33 @@ def setup_filter_parameters(report, querydict):
     return output
 
 
+def get_requested_ids(return_url_args):
+    return_url_querydict = QueryDict(return_url_args[1:])
+    activity = return_url_querydict.get('activity', None)
+    if activity:
+        requested_query = reconstruct_activity_query(return_url_args)
+        return list(map(int, requested_query.values_list('target_object_id', flat=True)))
+    else:
+        requested_query = reconstruct_query(return_url_args)
+        return list(requested_query.values_list('id', flat=True))
+
+
+def reconstruct_activity_query(next_qp):
+    """
+    Reconstruct the query filter on the previous page using the next
+    query parameter. note that if next is empty, the resulting query
+    will return all records.
+    """
+    querydict = QueryDict(next_qp)
+
+    _, selected_actions = dashboard_filter(querydict)
+    sort_expr, _ = activity_sort(querydict.getlist('sort'))
+    if selected_actions != []:
+        selected_actions = selected_actions.order_by(*sort_expr)
+
+    return selected_actions
```

Comment:
If I understand what's going on here, it looks like you're handling an edge case where selected actions _is_ empty?

If so, this approach might be more extensible in the case that you have more conditions / want to do more than just order later on:

```suggestion
    if not selected_actions:
        return selected_actions

    return selected_actions.order_by(*sort_expr)
```